src/internal/connector/data_collections.go

```diff
@@ -57,7 +58,7 @@ func (gc *GraphConnector) ProduceBackupCollections(
 		return nil, nil, clues.Stack(err).WithClues(ctx)
 	}
 
-	serviceEnabled, err := checkServiceEnabled(
+	serviceEnabled, deltaAvailable, err := checkServiceEnabled(
```

Comment:
naming nit: `deltaAvailable` sounds like you're flagging "we have a prior delta available to use".  A small fix would be to name it something like `mustNotDeltaQuery` or `canMakeDeltaQueries`.

A somewhat larger lift, butt one that might be more flexible, would be to retrieve the entire `api.UserInfo` here and pass it down (no strong feelings on how far to pass it down).  Adding a method like `userInfo.CanMakeDeltaQueries()` will allow us to establish an interface (if wanted), and will generally provide easy portability/extensibility by centralizing the ownership of delta-able identification within the api.  It also give us a userInfo struct in case we need to examine it for any other downstream/future case (mailbox settings, etc).  Finally, it exposes the check for SDK users to easily consume.

Either way works, I leave it up to you.

Reply:
I would say start with the low lift for now. I'm not sure adding more information to the backup path is what we want in the long run. There's already quite a bit going on

Reply:
I don't think we should pass down entire `userInfo` just yet. I was thinking we could do that once we have the requirement where we have to pass down more items which could also help us make more sense of how far to pass it down. I've made the other changes.